In [ ]:
print('Hello IE 507')

Hello IE 507


In [ ]:
!pip install -q pyomo

problem 1

$\min \ -3x_1-2x_2+8x_3-7x_4+0.19x_5+11x_6+20.2x_7 \\ \text{ s.t. }  \sum_{1}^{7}-x_i \geq -256 \\  \sum_{1}^{7}x_i \geq 256 \\ -x_2+x_3 \geq 10 \\ 3x1-x_5+12x_6 \geq -105 \\ 18.5x_2+30.5x_4-x_7 \geq 28 \\ -x_3-4x_5+x_7 \geq 31 \\ x_1 \geq -5,x_2 \geq0,x_3 \geq 5,x_4 \geq -3,x_5 \geq -4, \\x_6 \geq 0,x_7 \geq 10 $.  
---



In [ ]:
from pyomo.environ import *

In [ ]:
# create a model
model = ConcreteModel()

In [ ]:
# declare decision variables
model.x1 = Var(domain=Reals)
model.x2 = Var(domain=Reals)
model.x3 = Var(domain=Reals)
model.x4 = Var(domain=Reals)
model.x5 = Var(domain=Reals)
model.x6 = Var(domain=Reals)
model.x7 = Var(domain=Reals)



In [ ]:
# declare objective
model.cost = Objective(expr = -3*model.x1-2*model.x2+8*model.x3-7*model.x4+0.19*model.x5+11*model.x6+20.2*model.x7,sense=minimize)

In [ ]:
# declare constraints
model.constr1 = Constraint(expr = -model.x1-model.x2-model.x3-model.x4-model.x5-model.x6-model.x7>=-256)
model.constr2 = Constraint(expr = model.x1+model.x2+model.x3+model.x4+model.x5+model.x6+model.x7>=256)
model.constr3 = Constraint(expr = -model.x2+model.x3>=10)
model.constr4 = Constraint(expr = 3*model.x1-model.x5+12*model.x6>=-105)
model.constr5 = Constraint(expr = 18.5*model.x2+30.5*model.x4-model.x7>=28)
model.constr6 = Constraint(expr = -1*model.x3-4*model.x5+model.x7>=31)
model.constr7 = Constraint(expr = model.x1>=-5)
model.constr8 = Constraint(expr = model.x2>=0)
model.constr9 = Constraint(expr =model.x3>=5)
model.constr10 = Constraint(expr =model.x4>=-3)
model.constr11 = Constraint(expr =model.x5>=-4)
model.constr12 = Constraint(expr =model.x6>=0)
model.constr13 = Constraint(expr =model.x7>=10)


In [ ]:
model.pprint()

7 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  -5.0 :  None : False : False :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :   0.0 :  None : False : False :  Reals
    x3 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  10.0 :  None : False : False :  Reals
    x4 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None : 230.0 :  None : False : False :  Reals
    x5 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  -4.0 :  None : False : False :  Reals
    x6 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :   0.0 :  None : False : False :  Reals
    x7 : Size=1, Index=None
        Key  : Lower : Value            :

In [ ]:
!apt-get install -y -qq glpk-utils

In [ ]:
SolverFactory('glpk', executable='/usr/bin/glpsol').solve(model).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: -1010.76
  Upper bound: -1010.76
  Number of objectives: 1
  Number of constraints: 14
  Number of variables: 8
  Number of nonzeros: 33
  Sense: minimize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.012926101684570312
# ----------------------------------------------------------
#   Solution Information
# ----------------------------

In [ ]:
# display solution
print('\nCost = ', model.cost())

print('\nDecision Variables')
print('x1 = ', model.x1())
print('x2 = ', model.x2())
print('x3 = ', model.x3())
print('x4 = ', model.x4())
print('x5 = ', model.x5())
print('x6 = ', model.x6())
print('x7 = ', model.x7())

print('\nConstraints')
print('Constraint 1  = ', model.constr1())
print('Constraint 2  = ', model.constr2())
print('Constraint 3  = ', model.constr3())
print('Constraint 4  = ', model.constr4())
print('Constraint 5  = ', model.constr5())
print('Constraint 6  = ', model.constr6())
print('Constraint 7  = ', model.constr7())
print('Constraint 8  = ', model.constr8())
print('Constraint 9  = ', model.constr9())
print('Constraint 10  = ', model.constr10())
print('Constraint 11  = ', model.constr11())
print('Constraint 12  = ', model.constr12())
print('Constraint 13  = ', model.constr13())



Cost =  -1010.760000000002

Decision Variables
x1 =  -5.0
x2 =  0.0
x3 =  10.0
x4 =  230.0
x5 =  -4.0
x6 =  0.0
x7 =  24.9999999999999

Constraints
Constraint 1  =  -255.9999999999999
Constraint 2  =  255.9999999999999
Constraint 3  =  10.0
Constraint 4  =  -11.0
Constraint 5  =  6990.0
Constraint 6  =  30.9999999999999
Constraint 7  =  -5.0
Constraint 8  =  0.0
Constraint 9  =  10.0
Constraint 10  =  230.0
Constraint 11  =  -4.0
Constraint 12  =  0.0
Constraint 13  =  24.9999999999999


Answer:
Objective function value=-1010.760000000002 \\
the value decision variables at optimal point is as following: \\
x1 =  -5.0, \\
x2 =  0.0, \\
x3 =  10.0, \\
x4 =  230.0, \\
x5 =  -4.0, \\
x6 =  0.0, \\
x7 =  24.9999999999999 \\
Constraints activities: \\
1. constraint 1 is active since LHS = $ \sum_{i=1}^{7}x_i= -255.9999999999999 = RHS $ \\  
2. constraint 2 is active since LHS = $ \sum_{i=1}^{7}x_i= 255.9999999999999 = RHS $ \\
3. constraint 3 is active since LHS = $ -x_2+x_3 = 10 = RHS $ \\
4. constraint 4 is not active since LHS = $ 
 

In [ ]:
! pip install -q pyomo

     |████████████████████████████████| 9.6 MB 17.1 MB/s 
     |████████████████████████████████| 49 kB 5.9 MB/s 


Problem 2 \\
$\max \ 0.082x_1+0.036x_2+0.0156x_3+0.033x_4+0.054x_5+0.0125y \\ \text{s.t.} \ 3x_1-3x_2+5x_3-x_4+x_5-y \leq 80 \\ x_3+2x_4+4y \geq 30 \\ \frac{3x_1-1.5x_3+2x_4+1.8x_5+6x_6}{-x_2-x_4+x_5} \leq 3.6 \\
\frac{x_1-2x_2+3x_3+4x_4+5x_5}{-x_1+x_3+x_6-2y}\leq 6 \\ x_1,x_2,x_3,x_4,x_5,x_6 \geq 0 \\ y\in [-0.2,5.5] \\  $

$\huge{Assumption 1} \\ -x_2-x_4+x_5 \geq 0 , \\ \text{and} -x_1+x_3+x_6-2y \geq 0 $  
so now constraints are as following: \\
$ 3x_1-3x_2+5x_3-x_4+x_5-y \leq 80  \\ 
 x_3+2x_4+4y \geq 30$ \\
 $ 3x_1+3.6x_2-1.5x_3+5.6x_4-1.8x_5+6x_6 \leq 0 \\ 7x_1-2x_2-3x_3+4x_4+5x_5-6x_6+12y \leq 0 \\  x_1,x_2,x_3,x_4,x_5,x_6 \geq 0 \\ y\in [-0.2,5.5] \\   \\ -x_2-x_4+x_5 \geq 0 , \\ \text{and} -x_1+x_3+x_6-2y \geq 0 $  


In [ ]:
from pyomo.environ import *

In [ ]:
# create a model
model21 = ConcreteModel()

In [ ]:
# declare decision variables
model21.x1 = Var(domain=Reals)
model21.x2 = Var(domain=Reals)
model21.x3 = Var(domain=Reals)
model21.x4 = Var(domain=Reals)
model21.x5 = Var(domain=Reals)
model21.x6 = Var(domain=Reals)
model21.y = Var(domain=Reals)



In [ ]:
# declare objective
model21.cost = Objective(expr = 0.082*model21.x1+0.036*model21.x2+0.0156*model21.x3+0.033*model21.x4+0.054*model21.x5+0.0125*model21.y,sense=maximize)

In [ ]:
#  declare constraints
model21.constr1 = Constraint(expr = 3*model21.x1-3*model21.x2+5*model21.x3-model21.x4+model21.x5-model21.y <=80)
model21.constr2 = Constraint(expr = model21.x3 +2*model21.x4+4*model21.y>= 30)
model21.constr3 = Constraint(expr = 3*model21.x1 -1.5*model21.x3+2*model21.x4 +1.8*model21.x5+6*model21.x6 +3.6*model21.x2+3.6*model21.x4-3.6*model21.x5 <=0 )
model21.constr4 = Constraint(expr = model21.x1 -2*model21.x2 +3*model21.x3+ 4*model21.x4+5*model21.x5+6*model21.x1-6*model21.x3-6*model21.x6+12*model21.y <=0)
model21.constr6 = Constraint(expr = model21.x1>= 0)
model21.constr7 = Constraint(expr = model21.x2>=0)
model21.constr8 = Constraint(expr = model21.x3>= 0)
model21.constr9= Constraint(expr = model21.x4>= 0)
model21.constr10 = Constraint(expr = model21.x5>= 0)
model21.constr11= Constraint(expr = model21.y>= -0.2)
model21.constr12= Constraint(expr = model21.y<=5.5)
model21.constr13 = Constraint(expr = model21.x6>= 0)

This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().


In [ ]:
model21.pprint()

7 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :   0.0 :  None : False : False :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value            : Upper : Fixed : Stale : Domain
        None :  None : 18.0059553349874 :  None : False : False :  Reals
    x3 : Size=1, Index=None
        Key  : Lower : Value            : Upper : Fixed : Stale : Domain
        None :  None : 23.5593052109181 :  None : False : False :  Reals
    x4 : Size=1, Index=None
        Key  : Lower : Value            : Upper : Fixed : Stale : Domain
        None :  None : 3.22034739454106 :  None : False : False :  Reals
    x5 : Size=1, Index=None
        Key  : Lower : Value           : Upper : Fixed : Stale : Domain
        None :  None : 19.241687344913 :  None : False : False :  Reals
    x6 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :   0.0 :  None : False 

In [ ]:
!apt-get install -y -qq glpk-utils

In [ ]:
SolverFactory('glpk', executable='/usr/bin/glpsol').solve(model21).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: 2.12102483443709
  Upper bound: 2.12102483443709
  Number of objectives: 1
  Number of constraints: 13
  Number of variables: 8
  Number of nonzeros: 31
  Sense: maximize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.011226892471313477
# ----------------------------------------------------------
#   Solution Information
# ------------

In [ ]:
# display solution
print('\nCost = ', model21.cost())

print('\nDecision Variables')
print('x1 = ', model21.x1())
print('x2 = ', model21.x2())
print('x3 = ', model21.x3())
print('x4 = ', model21.x4())
print('x5 = ', model21.x5())
print('x6 = ', model21.x6())
print('x7 = ', model21.y())

print('\nConstraints')
print('Constraint 1  = ', model21.constr1())
print('Constraint 2  = ', model21.constr2())
print('Constraint 3  = ', model21.constr3())
print('Constraint 4  = ', model21.constr4())
print('Constraint 6  = ', model21.constr6())
print('Constraint 7  = ', model21.constr7())
print('Constraint 8  = ', model21.constr8())
print('Constraint 9  = ', model21.constr9())
print('Constraint 10  = ', model21.constr10())
print('Constraint 11  = ', model21.constr11())
print('Constraint 12  = ', model21.constr12())
print('Constraint 13  = ', model21.constr13())


Cost =  2.1210248344370837

Decision Variables
x1 =  0.0
x2 =  19.5529801324503
x3 =  24.2384105960265
x4 =  0.0
x5 =  18.9072847682119
x6 =  0.0
x7 =  1.44039735099338

Constraints
Constraint 1  =  80.00000000000011
Constraint 2  =  30.00000000000002
Constraint 3  =  -9.947598300641403e-14
Constraint 4  =  -4.618527782440651e-14
Constraint 6  =  0.0
Constraint 7  =  19.5529801324503
Constraint 8  =  24.2384105960265
Constraint 9  =  0.0
Constraint 10  =  18.9072847682119
Constraint 11  =  1.44039735099338
Constraint 12  =  1.44039735099338
Constraint 13  =  0.0


ANSWER  
Hence while taking both the restrictions \\
 $ -x_2-x_4+x_5 \geq 0 , \\ \text{and} -x_1+x_3+x_6-2y \geq 0 $  
 our Linear programming problem is as follows:
 $ 3x_1-3x_2+5x_3-x_4+x_5-y \leq 80  \\ 
 x_3+2x_4+4y \geq 30$ \\
 $ 3x_1+3.6x_2-1.5x_3+5.6x_4-1.8x_5+6x_6 \leq 0 \\ 7x_1-2x_2-3x_3+4x_4+5x_5-6x_6+12y \leq 0 \\  x_1,x_2,x_3,x_4,x_5,x_6 \geq 0 \\ y\in [-0.2,5.5] \\   \\ -x_2-x_4+x_5 \geq 0 , \\ \text{and} -x_1+x_3+x_6-2y \geq 0 $  

we get the optimal solution as the cost of  2.1210248344370837
and value of decision variables as following:
x1 =  0.0 \\
x2 =  19.5529801324503 \\
x3 =  24.2384105960265 \\
x4 =  0.0 \\
x5 =  18.9072847682119 \\
x6 =  0.0 \\
x7 =  1.44039735099338

In [ ]:
! pip install -q pyomo

     |████████████████████████████████| 9.6 MB 4.2 MB/s 
     |████████████████████████████████| 49 kB 5.4 MB/s 


Problem 3  
let jen industries produces  x kg quantity of product P and y  kg quantity of product Q now according the question 

$\min \ 1200x+2580y \ \\ \text{s.t.} \ x+y \geq 145 \\ 2.18x+\frac{13}{12}y \leq 980, \ x \geq 105 , \ y \geq 125 $

In [ ]:
from pyomo.environ import *

In [ ]:
# create a model
model3 = ConcreteModel()

In [ ]:
# declare decision variables
model3.x = Var(domain=Reals)
model3.y = Var(domain=Reals)

In [ ]:
# declare objective
model3.cost = Objective(expr = 1200*model3.x+2580*model3.y,sense=minimize)

In [ ]:
# declare constraints
model3.constr1 = Constraint(expr = model3.x+model3.y>=145) 
model3.constr2 = Constraint(expr = 26.16*model3.x + 13*model3.y <= 11760 )
model3.constr3 = Constraint(expr = model3.x >= 105)
model3.constr4 = Constraint(expr = model3.y >=125)

    'pyomo.core.base.constraint.ScalarConstraint'>) on block unknown with a
    new Component (type=<class
    'pyomo.core.base.constraint.AbstractScalarConstraint'>). This is usually
    indicative of a modelling error. To avoid this warning, use
    block.del_component() and block.add_component().
    'pyomo.core.base.constraint.ScalarConstraint'>) on block unknown with a
    new Component (type=<class
    'pyomo.core.base.constraint.AbstractScalarConstraint'>). This is usually
    indicative of a modelling error. To avoid this warning, use
    block.del_component() and block.add_component().
    'pyomo.core.base.constraint.ScalarConstraint'>) on block unknown with a
    new Component (type=<class
    'pyomo.core.base.constraint.AbstractScalarConstraint'>). This is usually
    indicative of a modelling error. To avoid this warning, use
    block.del_component() and block.add_component().
    'pyomo.core.base.constraint.ScalarConstraint'>) on block unknown with a
    new Component (ty

In [ ]:
model3.pprint()

2 Var Declarations
    x : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    y : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals

1 Objective Declarations
    cost : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : 1200*x + 2580*y

4 Constraint Declarations
    constr1 : Size=1, Index=None, Active=True
        Key  : Lower : Body  : Upper : Active
        None : 145.0 : x + y :  +Inf :   True
    constr2 : Size=1, Index=None, Active=True
        Key  : Lower : Body           : Upper   : Active
        None :  -Inf : 26.16*x + 13*y : 11760.0 :   True
    constr3 : Size=1, Index=None, Active=True
        Key  : Lower : Body : Upper : Active
        None : 105.0 :    x :  +Inf :   True
    constr4 : Size=1, Index=None, Active=True
      

In [ ]:
!apt-get install -y -qq glpk-utils

In [ ]:
SolverFactory('glpk', executable='/usr/bin/glpsol').solve(model3).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: 448500.0
  Upper bound: 448500.0
  Number of objectives: 1
  Number of constraints: 5
  Number of variables: 3
  Number of nonzeros: 7
  Sense: minimize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.01113748550415039
# ----------------------------------------------------------
#   Solution Information
# -------------------------------

In [ ]:
# display solution
print('\nCost = ', model3.cost())

print('\nDecision Variables')
print('x = ', model3.x())
print('y = ', model3.y())
print('\nConstraints')
print('Constraint 1  = ', model3.constr1())
print('Constraint 2  = ', model3.constr2())
print('Constraint 3  = ', model3.constr3())
print('Constraint 4  = ', model3.constr4())


Cost =  448500.0

Decision Variables
x =  105.0
y =  125.0

Constraints
Constraint 1  =  230.0
Constraint 2  =  4371.8
Constraint 3  =  105.0
Constraint 4  =  125.0


Answer: \\
1. let  Jen Industries produces x kgs of product P, \\
   and produces y kgs of product Q, \\
   now according to question, since minimum combined production of both products P and Q is 145 kgs so "$ x+y \geq 145 $" \\
   now since  major customer has placed an order for 105
Kgs of product P and 125 Kgs of product Q so industries has to produce minimum 105 kgs of product P and 125 kgs of product Q so \\
$ "x \geq 105" , \ "y \geq 125" $ \\
since Product P requires 2.18 hours of manufacturing
time per Kg and product Q requires 65 minutes of manufacturing time per Kg
and given that 980 hours are availabe for manufacturing so \\
$ 2.18x+(13/12)y \leq 980 \\ 26.16x+13y \leq 11760 $ \\
2. optimal cost: 448500.0 \\
   value of decision variable:
   x =  105.0,
y =  125.0 \\
   Activeness of constraints: \\
   constraint 1 is not active since LHS= x+y = 105+125 =230 which is not equal to RHS. \\
   Constraint 2 is not active since LHS = 4371 which is not equal to RHS \\
   constraint 3 is active since LHS = 105 which is equal to RHS \\
   constraint 4 is active since LHS = 125 which is equal to RHS \\


Answer 3:  
Since customer has placed order of 42 kgs of pruduct P and 97 kgs of pruduct Q.so now for insdusries it is required to pruduce minimum 147kgs of pruduct P and  222 kgs of pruduct Q. so now Constraint 3, constraint 4 of previous Linear Programming problem are changed to following:
$ "x \geq 147 ", "x \geq 222"$
now the new changed Linear programming problem is as follows:
$\min \ 1200x+2580y \\ \text{s.t.} \ x+y \geq 145 \\ 2.18x+\frac{13}{12}y \leq 980, \ x \geq 147 , \ y \geq 222 $

In [ ]:
! pip install -q pyomo


In [ ]:
from pyomo.environ import *

In [ ]:
# create a model
model4 = ConcreteModel()

In [ ]:
# declare decision variables
model4.x = Var(domain=Reals)
model4.y = Var(domain=Reals)

In [ ]:
# declare objective
model4.cost = Objective(expr = 1200*model4.x+2580*model4.y,sense=minimize)

    'pyomo.core.base.objective.ScalarObjective'>) on block unknown with a new
    Component (type=<class 'pyomo.core.base.objective.ScalarObjective'>). This
    is usually indicative of a modelling error. To avoid this warning, use
    block.del_component() and block.add_component().


In [ ]:
# declare constraints
model4.constr1 = Constraint(expr = model4.x+model4.y>=145) 
model4.constr2 = Constraint(expr = 26.16*model4.x + 13*model4.y <= 11760 )
model4.constr3 = Constraint(expr = model4.x >= 147)
model4.constr4 = Constraint(expr = model4.y >=222)

    'pyomo.core.base.constraint.ScalarConstraint'>) on block unknown with a
    new Component (type=<class
    'pyomo.core.base.constraint.AbstractScalarConstraint'>). This is usually
    indicative of a modelling error. To avoid this warning, use
    block.del_component() and block.add_component().
    'pyomo.core.base.constraint.ScalarConstraint'>) on block unknown with a
    new Component (type=<class
    'pyomo.core.base.constraint.AbstractScalarConstraint'>). This is usually
    indicative of a modelling error. To avoid this warning, use
    block.del_component() and block.add_component().
    'pyomo.core.base.constraint.ScalarConstraint'>) on block unknown with a
    new Component (type=<class
    'pyomo.core.base.constraint.AbstractScalarConstraint'>). This is usually
    indicative of a modelling error. To avoid this warning, use
    block.del_component() and block.add_component().
    'pyomo.core.base.constraint.ScalarConstraint'>) on block unknown with a
    new Component (ty

In [ ]:
model4.pprint()

2 Var Declarations
    x : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    y : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals

1 Objective Declarations
    cost : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : 1200*x + 2580*y

4 Constraint Declarations
    constr1 : Size=1, Index=None, Active=True
        Key  : Lower : Body  : Upper : Active
        None : 145.0 : x + y :  +Inf :   True
    constr2 : Size=1, Index=None, Active=True
        Key  : Lower : Body           : Upper   : Active
        None :  -Inf : 26.16*x + 13*y : 11760.0 :   True
    constr3 : Size=1, Index=None, Active=True
        Key  : Lower : Body : Upper : Active
        None : 147.0 :    x :  +Inf :   True
    constr4 : Size=1, Index=None, Active=True
      

In [ ]:
!apt-get install -y -qq glpk-utils

In [ ]:
SolverFactory('glpk', executable='/usr/bin/glpsol').solve(model4).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: 749160.0
  Upper bound: 749160.0
  Number of objectives: 1
  Number of constraints: 5
  Number of variables: 3
  Number of nonzeros: 7
  Sense: minimize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.01419830322265625
# ----------------------------------------------------------
#   Solution Information
# -------------------------------

In [ ]:
# display solution
print('\nCost = ', model4.cost())

print('\nDecision Variables')
print('x = ', model4.x())
print('y = ', model4.y())
print('\nConstraints')
print('Constraint 1  = ', model4.constr1())
print('Constraint 2  = ', model4.constr2())
print('Constraint 3  = ', model4.constr3())
print('Constraint 4  = ', model4.constr4())


Cost =  749160.0

Decision Variables
x =  147.0
y =  222.0

Constraints
Constraint 1  =  369.0
Constraint 2  =  6731.52
Constraint 3  =  147.0
Constraint 4  =  222.0


 Answer 4: 
 optimal cost: 749160.0
 \\
   value of decision variable:
   x =  147.0,
y =  222.0 \\
   Activeness of constraints: \\
   constraint 1 is not active since LHS= x+y = 147+222 =369 which is not equal to RHS. \\
   Constraint 2 is not active since LHS = 6731.52 which is not equal to RHS \\
   constraint 3 is active since LHS = 147 which is equal to RHS \\
   constraint 4 is active since LHS = 222 which is equal to RHS \\
   after the order, jen industries has to increase there productity of both product P and Q.
   for optimal cost , Insdustries has to produce 147 kgs of pruduct P and 222 kgs of product Q.

